lib/backup_restore/backup_restore.rb

```diff
@@ -60,11 +60,30 @@ def self.can_rollback?
     backup_tables_count > 0
   end
 
+  def self.strict_backup_creation?
+    ENV["DISCOURSE_STRICT_BACKUP_CREATION"] == "1" && Discourse.store.external?
+  end
+
+  def self.disallow_backups!
+    $redis.set(backup_disallowed_key, 1)
+  end
+
+  def self.backups_disallowed?
+    $redis.get(backup_disallowed_key).present?
```

Comment:
Another trivial change but redis has an `exists` method which is slightly simpler.